NewYearGift/NewYearGift/Services/UserAction.cs

```diff
@@ -0,0 +1,82 @@
+﻿using NewYearGift.Models;
+using NewYearGift.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization.Formatters;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NewYearGift.Services
+{
+    internal class UserAction
+    {
+        private GiftBox giftBox = GiftBox.GetInstance;
+        private SortSweets sortSweets = new SortSweets();
```

Comment:
GiftBox and SortSweet should be in one class, as we are working with one business unit 'Sweet'